h/accounts/__init__.py

```diff
@@ -34,8 +35,25 @@ def make_staff(username):
         raise NoSuchUserError
 
 
+def authenticated_user(request):
+    """Return the authenticated user or None.
+
+    :rtype: h.accounts.models.User or None
+
+    """
+    if not request.authenticated_userid:
+        return None
+
+    # pylint: disable=unpacking-non-sequence
+    username, _ = util.split_user(request.authenticated_userid)
```

Comment:
I'd like us to check that the second half of this tuple matches `request.domain`.